libminifi/include/utils/MinifiConcurrentQueue.h

```diff
@@ -29,11 +29,12 @@ namespace nifi {
 namespace minifi {
 namespace utils {
 
+
+// Provides a queue API and guarantees no race conditions in case of multiple producers and consumers.
 template <typename T>
 class ConcurrentQueue {
  public:    
-  ConcurrentQueue() = default;
-  virtual ~ConcurrentQueue() = default;
+  explicit ConcurrentQueue() = default;
```

Comment:
I don't see the need to make this explicit, although it doesn't hurt much. It prevents code like this:
`ConcurrentQueue make_queue() { return {}; }`